### DISCUSSION TITLE
SRT file translation using ChatGPT automation

### DISCUSSION THREAD

**Discussion by EtienneAb3d** [openai/whisper#1943] [Show and tell]
SRT file translation may be complex because of possible bad sentence cut, making each text part loosing its context.

A Whisper user reported me that ChatGPT, with its ability to handle complex processing, is producing much better SRT translations than any other available tools he tested. He asked me if I could automate file processing for this translation task.

I just wrote a versatile Java application to automate ChatGPT queries on a set of files, and I'm providing the SRT translation as an example use case (see the ReadMe):
https://github.com/EtienneAb3d/ChatMate

:)

**Comment by Zayno** [comment]
Thank you! have you looked at "Subtitle Edit" ?. it's an ALL-In-One open source subtitle software. if your software adds any new features I'd like to see it in Subtitle Edit as well!

**Reply by EtienneAb3d** [reply]
@Zayno 

As far as I understand it, Subtitle Edit is using Google Translate. Probably, it is translating each piece of text, timestamp by timestamp, loosing the context. Translating the SRT file as a whole, using advance capabilities of ChatGPT, certainly brings with much better results.

This said, Subtitle Edit doesn't really need my ChatMate software dedicated to ChatGPT file processing automation, but only the few lines of code needed to call the ChatGPT API with the prompt provided as an example for the SRT translation. You should ask this to the Subtitle Edit creator.
;-)
